pkg/api/v1/types.go

```diff
@@ -1649,6 +1649,10 @@ type Endpoints struct {
 type EndpointSubset struct {
 	// IP addresses which offer the related ports.
 	Addresses []EndpointAddress `json:"addresses,omitempty"`
+	// IP addresses which offer the related ports but are not currently marked as ready
+	// because they have not yet finished starting, or have recently failed a liveness
```

Comment:
Failing a liveness check will result in ready being cleared as well.  I'll add both.